diff -Nrup kexec-tools-2.0.0.orig/kexec/kexec-syscall.h kexec-tools-2.0.0/kexec/kexec-syscall.h
--- kexec-tools-2.0.0.orig/kexec/kexec-syscall.h	2010-08-05 15:21:47.000000000 +0800
+++ kexec-tools-2.0.0/kexec/kexec-syscall.h	2010-08-05 15:23:41.000000000 +0800
@@ -88,6 +88,6 @@ static inline long kexec_reboot(void)
 #define KEXEC_ARCH_MIPS_LE (10 << 16)
 #define KEXEC_ARCH_MIPS    ( 8 << 16)
 
-#define KEXEC_MAX_SEGMENTS 16
+#define KEXEC_MAX_SEGMENTS 128
 
 #endif /* KEXEC_SYSCALL_H */
